tests/lib/bin/version-tool

```diff
@@ -0,0 +1,413 @@
+    # type: (Text, int) -> bool
+    """
+    consistent_relation returns true if the relation is consistent with delta.
+
+    The relation operator is one of ==, !=, <, <=, > or >=.
+    Delta is either 0, a positive or a negative number.
+    """
+    if rel_op == "==":
+        return delta == 0
+    elif rel_op == "!=":
+        return delta != 0
+    elif rel_op == ">":
+        return delta > 0
+    elif rel_op == ">=":
+        return delta >= 0
+    elif rel_op == "<":
+        return delta < 0
+    elif rel_op == "<=":
+        return delta <= 0
+    raise ValueError("unexpected relational operator " + rel_op)
+
+
+class ConsistentRelationTests(unittest.TestCase):
+    def test_eq(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation("==", -1))
+        self.assertTrue(consistent_relation("==", 0))
+        self.assertFalse(consistent_relation("==", +1))
+
+    def test_ne(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("!=", -1))
+        self.assertFalse(consistent_relation("!=", 0))
+        self.assertTrue(consistent_relation("!=", +1))
+
+    def test_gt(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation(">", -1))
+        self.assertFalse(consistent_relation(">", 0))
+        self.assertTrue(consistent_relation(">", +1))
+
+    def test_ge(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation(">=", -1))
+        self.assertTrue(consistent_relation(">=", 0))
+        self.assertTrue(consistent_relation(">=", +1))
+
+    def test_lt(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<", -1))
+        self.assertFalse(consistent_relation("<", 0))
+        self.assertFalse(consistent_relation("<", +1))
+
+    def test_le(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<=", -1))
+        self.assertTrue(consistent_relation("<=", 0))
+        self.assertFalse(consistent_relation("<=", +1))
+
+    def test_unknown(self):
+        # type: () -> None
+        with self.assertRaises(ValueError):
+            consistent_relation("???", 0)
+
+
+def numeric_part_of(text):
+    # type: (Text) -> int
+    """
+    numeric_part_of returns the integer representing the numeric portion of text.
+
+    If text does not start with an natural number the result is zero. The
+    function never fails. Negative numbers are not supported because they are
+    meaningless for version numbers.
+    """
+    match = re.match("^([0-9]+)", text)
+    if match:
+        return int(match.group(1))
+    return 0
+
+
+class NumericPartOfTests(unittest.TestCase):
+    def test_simple(self):
+        # type: () -> None
+        self.assertEqual(numeric_part_of("12"), 12)
+        self.assertEqual(numeric_part_of("12-stuff"), 12)
+        self.assertEqual(numeric_part_of("stuff-12"), 0)
+        self.assertEqual(numeric_part_of("-12"), 0)
+
+
+def naive_version_cmp(a, b):
```

Comment:
We don't need this anymore with "strict" mode, right? I would like to remove it (we can keep it in the history) so that its not used "accidentally" (and then we may run into the concerns I have with this lenient version).

Reply:
I meant to keep it but if you feel strongly about it I will remove it now.